sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/SwitchSectionShouldNotHaveTooManyStatements_DefaultValue.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public class Program
+    {
+        public Program()
+        {
+            switch (myVariable)
+            {
+                case 0:
+                    break;
+                case 1: // Noncompliant {{Reduce this 'switch/case' number of lines from 6 to at most 5, for example by extracting code into a method.}}
+//              ^^^^^^^
+                    Console.WriteLine("1");
+                    Console.WriteLine("2");
+                    Console.WriteLine("3");
+                    Console.WriteLine("4");
+                    Console.WriteLine("5");
+                    break;
+                case 2:
+                    {
+                        Console.WriteLine("1");
+                        Console.WriteLine("2");
+                        Console.WriteLine("3");
+                        Console.WriteLine("4");
+                        break;
+                    }
+                case 3: // Noncompliant {{Reduce this 'switch/case' number of lines from 6 to at most 5, for example by extracting code into a method.}}
+                case 4:
+                    if (true)
+                    {
+                        if (false)
+                        {
+                            for (int i = 0; i < length; i++)
+                            {
+                                Console.WriteLine();
+                                Console.WriteLine();
+                                Console.WriteLine();
+                            }
+                        }
+                    }
+                case 5: // Noncompliant
+                    var x = Foo(42);
+                    break;
+
+                    string Foo(int value)
```

Comment:
Not sure what to do with local functions. Shall we ignore them as we recommend to use a method? Shall I skip the function declaration (e.g. `string Foo(int value)`) from the count?

Reply:
The rule is about readability - blocks with too many lines are difficult to read, we should not ignore locals